src/robusta/integrations/kubernetes/custom_crds.py

```diff
@@ -0,0 +1,130 @@
+from typing import ClassVar
+from pydantic.dataclasses import dataclass
+from hikaru.model.rel_1_26 import HikaruDocumentBase, ObjectMeta, LabelSelector
+import datetime
+import pytz
+from kubernetes import client
+from hikaru.crd import HikaruCRDDocumentMixin
+
+
+@dataclass
+class CRDBase:
+    plural: ClassVar[str] = ""
+    group: ClassVar[str] = ""
+    version: ClassVar[str] = ""
+    total_pods_path: ClassVar[str] = ""
+    ready_pods_path: ClassVar[str] = ""
+
+    metadata: ObjectMeta
+    spec: dict
+    status: dict
+    apiVersion: str
+    kind: str
+
+    @classmethod
+    def readNamespaced(cls, name: str, namespace: str):
+        obj = client.CustomObjectsApi().get_namespaced_custom_object(
+            cls.group, cls.version, namespace, cls.plural, name
+        )
+        obj = cls(**obj)
+        return type("", (object,), {"obj": obj})()
+
+    @classmethod
+    def rollout_restart_patch(cls):
+        return {
+            "metadata": {
+                "annotations": {
+                    "kubectl.kubernetes.io/restartedAt": datetime.datetime.utcnow()
+                    .replace(tzinfo=pytz.UTC)
+                    .isoformat()
+                }
+            }
+        }
+
+    @classmethod
+    def list_namespaced(cls, namespace: str):
+        crd_res = client.CustomObjectsApi().list_namespaced_custom_object(
+            group=cls.group,
+            version=cls.version,
+            namespace=namespace,
+            plural=cls.plural,
+        )
+        crd_list = type(
+            "",
+            (object,),
+            {"items": [cls(**crd) for crd in crd_res.get("items", [])]},
+        )()
+
+        return crd_list
+
+    @classmethod
+    def list_for_all_namespaces(cls):
+        crd_res = client.CustomObjectsApi().list_cluster_custom_object(
+            group=cls.group,
+            version=cls.version,
+            plural=cls.plural,
+        )
+        crd_list = type(
+            "",
+            (object,),
+            {"items": [cls(**crd) for crd in crd_res.get("items", [])]},
+        )()
+
+        return crd_list
+
+
+# https://github.com/orgs/strimzi/discussions/9140 // restarts
+# https://github.com/strimzi/strimzi-kafka-operator/blob/main/helm-charts/helm3/strimzi-kafka-operator/crds/042-Crd-strimzipodset.yaml
+@dataclass
+class StrimziPodSet(CRDBase, HikaruDocumentBase, HikaruCRDDocumentMixin):
```

Comment:
Looks like the distance to create something completely configurable is very very small!